lib/sqlalchemy/sql/elements.py

```diff
@@ -4155,7 +4267,7 @@ def over(self, partition_by=None, order_by=None, range_=None, rows=None):
         )
 
     @util.memoized_property
-    def type(self):
+    def type(self) -> TypeEngine[_T]:
         wgt = self.element.within_group_type(self)
         if wgt is not None:
             return wgt
```

Comment:
`within_group_type()` returns `None` by default, it's likely meant to be overridden - so mypy infers `Any` for `wgt` here.